src/MOI_wrapper.jl

```diff
@@ -596,31 +598,35 @@ function backward_conic!(model::Optimizer, dA::Array{Float64,2}, db::Array{Float
         z = (x, y - s, [1.0])
         u, v, w = z
 
-        Q = [
-            zeros(n,n)   A'           c;
-            -A           zeros(m,m)   b;
-            -c'          -b'          zeros(1,1)
-        ]
+        # reshaping vectors to matrix
+        b = b[:, :]
+        c = c[:, :]
+
+        Q = sparse([
+            spzeros(n,n)   A'           c;
+            -A           spzeros(m,m)   b;
+            -c'          -b'          spzeros(1,1)
+        ])
 
         # find gradient of projections on dual of the cones
         Dπv = Dπ([MOI.dual_set(cone) for cone in cones], v)
 
-        M = ((Q - Matrix{Float64}(I, size(Q))) * BlockDiagonal([Matrix{Float64}(I, length(u), length(u)), Dπv, Matrix{Float64}(I,1,1)])) + Matrix{Float64}(I, size(Q))
+        M = ((Q - sparse(1.0I, size(Q))) * blockdiag(sparse(1.0I, length(u), length(u)), sparse(Dπv), sparse(1.0I, 1, 1))) + sparse(1.0I, size(Q))
 
         # find projections on dual of the cones
-        πz = vcat([u, π([MOI.dual_set(cone) for cone in cones], v), max.(w,0.0)]...)
+        πz = sparse(vcat([u, π([MOI.dual_set(cone) for cone in cones], v), max.(w,0.0)]...))
 
-        dQ = [
-            zeros(n,n)    dA'          dc;
-            -dA           zeros(m,m)   db;
-            -dc'         -db'          zeros(1,1)
-        ]
+        dQ = sparse([
+            spzeros(n,n)    dA'          dc;
+            -dA           spzeros(m,m)   db;
+            -dc'         -db'          spzeros(1,1)
+        ])
```

Comment:
this part is doing much conversion work.
```
[
    a  b  c
    a  b  c
]
```

creates a dense matrix out of the sparse arrays, which is then re-converted to sparse.
It would be better to create an empty dQ with spzeros(n+m+1, n+m+1) and then fill the non-diagonal blocks
with
dQ[n+1:n+1+m,1:n] .= -dA
etc.

Same thing above for the creation of Q

Reply:
oh. thanks for pointing out. never realized this redundant sparse-dense-sparse conversion

Reply:
I think the spzeros(1,1) can be replaced by just `0`

Reply:
fixed